src/extension/services/validationService/checks/xcode.ts

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import * as nls from "vscode-nls";
+import { basicCheck, createVersionErrorMessage, parseVersion } from "../util";
+import { IValidation, ValidationCategoryE, ValidationResultT } from "./types";
+
+nls.config({
+    messageFormat: nls.MessageFormat.bundle,
+    bundleFormat: nls.BundleFormat.standalone,
+})();
+const toLocale = nls.loadMessageBundle();
+
+const label = "xcodeVersion";
+
+async function test(): Promise<ValidationResultT> {
+    const command = "xcodebuild -version";
+    const result = await basicCheck({
+        command: command,
+        getVersion: parseVersion.bind(null, command, /\d+\.\d+\.\d+/gi),
+        versionRange: "11.3.1",
+    });
+    if (result.exists) {
+        if (result.versionCompare === -1) {
+            return {
+                status: "partial-success",
+                comment:
+                    "Detected version is older than 11.3.1 " +
+                    "Please update xcode in case of errors",
```

Comment:
I think in the comments you can write the names of products with a capital letter